charmhub_lp_tools/charm_project.py

```diff
@@ -333,33 +352,87 @@ def ensure_charm_recipes(self, branches: Optional[List[str]] = None,
+                        project=self.lp_project,
+                        store_name=self.charmhub_name,
+                        channels=build_from['channels'])
+                    print(f'Created charm recipe {lp_recipe.name}')
 
             else:
                 print(f'No changes needed for charm recipe {recipe_name}')
 
-        # TODO (wolsen) Check to see if there are any remaining charm recipes
-        #  configured in Launchpad and remove them (?). Remaining charm recipes
-        #  will be those left in the charm_lp_recipe_map dict. Not doing this
-        #  currently as its not clear that we want to remove them automatically
-        #  (yet).
+        # If remove_unknown option is used, then delete the unknown recipes.
+        if remove_unknown and current['non_config_recipes']:
+            for recipe_name in current['non_config_recipes'].keys():
+                if dry_run:
+                    print(f'Would delete {recipe_name} (dry_run)')
+                else:
+                    self.lpt.delete_charm_recipe_by_name(
+                        recipe_name,
+                        self.lp_team,
+                        self.lp_project)
+
+    def delete_recipe_by_name(self,
+                              recipe_name: str,
+                              dry_run: bool = True,
+                              ) -> None:
+        """Delete a recipe filtered by it's full name.
+
+        :param recipe_name: the recipe name
+        :raises KeyError: if the recipe couldn't be found.
+        """
+        if dry_run:
+            print(f'Would delete {recipe_name} (dry_run)')
+        else:
+            self.lpt.delete_charm_recipe_by_name(
+                recipe_name,
+                self.lp_team,
+                self.lp_project)
+
+    def delete_recipe_by_branch_and_track(self,
+                                          branch: str,
+                                          track: str,
+                                          dry_run: bool = True,
+                                          ) -> None:
+        """Delete a recipe filtered by track and risk.
+
+        If the recipe doesn't exist a warning is printed.
+
+        :param branch: the branch to delete
+        :param track: the track to delete.
+        :raises KeyError: if the recipe couldn't be found.
+        """
+        branch_name = branch.replace('/', '-')
+        recipe_name = DEFAULT_RECIPE_FORMAT.format(
+            project=self.lp_project.name,
+            branch=branch_name,
+            track=track)
+        if dry_run:
+            print(f'Would delete {recipe_name} (dry_run)')
+        else:
+            self.lpt.delete_charm_recipe_by_name(
+                recipe_name,
+                self.lp_team,
+                self.lp_project)
 
-    def _calc_recipes_for_repo(self, filter_by: Optional[List[str]] = None,
+    def _calc_recipes_for_repo(self,
+                               filter_by: Optional[List[str]] = None,
```

Comment:
add filter_by to the docstring

Reply:
It's already present; is it not sufficient?